lib/Optimizer/Transforms/LowerToCFGPatterns.inc

```diff
@@ -0,0 +1,104 @@
+/*******************************************************************************
+ * Copyright (c) 2022 - 2025 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+// These patterns are used by the lower-to-cfg pass and cc-loop-unroll pass.
+
+// This file must be included after a `using namespace mlir;` as it uses bare
+// identifiers from that namespace.
+
+namespace {
+class RewriteIf : public OpRewritePattern<cudaq::cc::IfOp> {
+public:
+  using OpRewritePattern::OpRewritePattern;
+
+  explicit RewriteIf(MLIRContext *ctx)
+      : OpRewritePattern(ctx), rewriteOnlyIfConst(false) {}
```

Comment:
I don't think you need this if you add a default initializer as suggested below?